src/form-elements/FormElementRepeatableSet.tsx

```diff
@@ -109,8 +109,16 @@ function FormElementRepeatableSet({
     formElementConditionallyShown.type === 'repeatableSet'
       ? formElementConditionallyShown.entries
       : {}
+
+  let validationClass = ''
+  if (displayValidationMessage) {
+    validationClass = formElementValidation
+      ? 'ob-repeatable-set-element__invalid'
+      : 'ob-repeatable-set-element__valid'
+  }
```

Comment:
Could do 
```
const validationClass = displayValidationMessage && formElementValidation ? 'ob-repeatable-set-element__invalid' : 'ob-repeatable-set-element__valid'
```

Up to you though

Reply:
Dave, your logic will result in no class unless `displayValidationMessage` is `true`.

Zac's logic will result in a class always being added.

The section element works like Zac's proposal so I would suggested implementing it (if the section element logic is what we are going for)

Reply:
I don't like the idea of showing the valid class when the user has not done anything.  Which is the way section elements work, although the implementation may be a bit different.  When the form opens there should be neither valid nor invalid class on the element.

Reply:
ie section elements have no classes when the form loads

Reply:
The section element will only show the `valid` class if everything in the section is valid. If you navigate to the form and the section element contains required elements that have no value, it will not have `valid` class. It will also not have the `invalid` class until `displayValidationMessage` is true (which is trigger but navigating to another page, pressing submit, or collapsing the section). Thinking this through, I don't think either of your proposals matches the section element logic 🤔.

Reply:
That sounds like Dave's code would do just that.

Reply:
Or is it not doing this part 

> If you navigate to the form and the section element contains required elements that have no value, it will not have valid class.

Reply:
Dave's code will do it

Reply:
Daves code does this:

> The `valid` class will only be applied if `displayValidationMessage` is true (which is triggered by the user navigating to another page or pressing submit)

The logic we want is

> The `valid` class will be applied if everything in the section is valid (which is determined by `formElementValidation` not having a value)

Reply:
I believe the logic we want could be implemented like so:

```
const validationClass = React.useMemo(() => {
  if (!formElementValidation) {
    return 'ob-repeatable-set-element__valid'
  }
  if (displayValidationMessage) {
    return 'ob-repeatable-set-element__invalid'
  }
}, [])
```

Reply:
Ah yeah, better update the others then.

Reply:
Sounds like a job for a hooky hook.